Data on herbicide use efficiency in cassava production.

mitchynjukuya opened the pull request.

rhijmans commented:
Thanks for doing this tricky one. 

Where did you get the key to herbicide codes? (it always good to add that type of information as a comment to the script). I am asking because there are a few left: Fa, NP, Zpo, ZPo.  Some of these you had set to "none", but that does not seem right. Did you mean "I do not know"?  I fixed a few of your changes where you used the *product name* instead of the *active ingredients*. I have also added missing active ingredients to the standard.   

The "notes" field is for the end-users to alert them, for example, about possible error or more data that could be extracted. In this case these were more things that we need to fix first. 

There are a few more variables that need to be extracted, including variables where there are multiple observations for an experimental unit during the growing season. These go in a separate file "timerecs". That is a bit more work and I cannot do it right now, so I put the file into "_pending" so that we do not forget. You can look at other examples in the pest_disease folder if you want to give it a try.

mitchynjukuya commented:
I got the key for herbicide codes in the raw data under variables Pretreat
and Posttreat (which l assume mean Pre-emergent and Post-emergent
herbicides respectively). The values l set to "none" initially had Fa_Prt
or NP_ZPo or NP_Zpo under treatment, Farmer's practice and/or Untreated
under both  Pretreat and Posttreat. "none" was to mean no herbicide was
applied. Will have a look at other examples as per your suggestions.

On Tue, Oct 8, 2024 at 6:43 PM Robert Hijmans ***@***.***>
wrote:

> Thanks for doing this tricky one.
>
> Where did you get the key to herbicide codes? (it always good to add that
> type of information as a comment to the script). I am asking because there
> are a few left: Fa, NP, Zpo, ZPo. Some of these you had set to "none", but
> that does not seem right. Did you mean "I do not know"? I fixed a few of
> your changes where you used the *product name* instead of the *active
> ingredients*. I have also added missing active ingredients to the
> standard.
>
> The "notes" field is for the end-users to alert them, for example, about
> possible error or more data that could be extracted. In this case these
> were more things that we need to fix first.
>
> There are a few more variables that need to be extracted, including
> variables where there are multiple observations for an experimental unit
> during the growing season. These go in a separate file "timerecs". That is
> a bit more work and I cannot do it right now, so I put the file into
> "_pending" so that we do not forget. You can look at other examples in the
> pest_disease folder if you want to give it a try.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/reagro/carob/pull/557#issuecomment-2400356025>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBUBPUDXAVPFDJF5LXVDKZLZ2QDRZAVCNFSM6AAAAABPRPGUP2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBQGM2TMMBSGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

rhijmans commented:
Thanks, I see, you were right. 

In this case, it seems to me that it would be better to use the herbicide names from "Pretreat" and "Posttreat" instead of the derived codes.  "Untreated" is much clearer than "NP_ZPo". Also, is "FarmerPractise" the same as "none"? I have no way of knowing, so I would set that to `NA`.  

I also think we should keep the "Pretreat" and "Posttreat" herbicides as separate variables. The rule is that we capture all the important variables from the experiment; the standard may need to be adjusted to be able to accommodate that.
